src/flag_gems/ops/masked_fill.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+import torch
+import triton
+import triton.language as tl
+
+from ..utils import broadcastable_to, libentry
+
+
+def cfggen():
+    block_m = [1, 2, 4, 8]
+    configs = [
+        triton.Config({"BLOCK_M": m, "BLOCK_N": 128}, num_warps=4) for m in block_m
+    ]
+    return configs
+
+
+@libentry()
+@triton.autotune(configs=cfggen(), key=["M", "N"])
+@triton.jit
+def masked_fill_kernel(
+    inp, expand_mask, value, out, M, N, BLOCK_M: tl.constexpr, BLOCK_N: tl.constexpr
+):
+    pid = tl.program_id(0)
+    rows_offset = pid * BLOCK_M + tl.arange(0, BLOCK_M)[:, None]
+    rows_mask = rows_offset < M
+
+    for off in range(0, N, BLOCK_N):

```

Comment:
Why loop over dimension `N` if there is no actual dependencies between iterations?

Reply:
Umm... I VAGUELY remember that this loop exists not just to solve the dependencies between iterations. 

My understanding is that if we don't introduce it, we would need the TILE to cover the entire row, specifying BLOCK_N as the next_power_of_2 for N. However, in Triton, which has limits on the return values of next_power_of_2, we might run into issues if that number isn't large enough. Also, it may increase the register usage...?  

If this isn't something that needs to be addressed here, please let me know, and I can adjust that :)